### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by tsimafeikhrol**

### CODE REVIEW

File: bucket.tf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 data "aws_caller_identity" "current" {}
 
 resource "aws_s3_bucket" "build_cache" {
-  bucket = "${data.aws_caller_identity.current.account_id}-gitlab-runner-cache-test"
+  bucket = "${data.aws_caller_identity.current.account_id}-${var.environment}-cache"
```
Comment (ToROxI) [inline bucket.tf]: maybe better to name `${data.aws_caller_identity.current.account_id}-${var.environment}-gitlab-cache`?

File: main.tf [hunk 2]
```diff
@@ -21,8 +22,7 @@ resource "aws_security_group" "runner" {
     from_port   = 22
     to_port     = 22
     protocol    = "tcp"
-    #cidr_blocks = ["${var.bastion_ip}/32"] # only from bastion
-    cidr_blocks = ["0.0.0.0/0"] # only from bastion
+    cidr_blocks = ["${var.bastion_ip}/32"]
```
Comment (ToROxI) [inline main.tf]: Looks like `bastion_ip` becomes a mandatory parameter for module. But what if I don't have or don't need bastion instance at all?

File: variables.tf [hunk 3]
```diff
@@ -1,49 +1,23 @@
...
 
-variable "availability_zone_proxy" {
-  description = "Availability zone used to host the proxy."
-  type        = "string"
-  default     = "a"
-}
-
-variable "subnet_id_gitlab_runner" {
-  description = "Subnet used for hosting the gitlab-runner."
+variable "aws_region" {
```
Comment (ToROxI) [inline variables.tf]: why this module contains variable to configure AWS Region?

File: variables.tf [hunk 4]
```diff
@@ -1,49 +1,23 @@
...
+  description = "AWS region."
   type        = "string"
 }
 
-variable "instance_type" {
-  description = "Instance type used for the gitlab-runner."
+variable "cache_user" {
```
Comment (ToROxI) [inline variables.tf]: This variable has no any references in code

File: variables.tf [hunk 5]
```diff
@@ -56,20 +30,37 @@ variable "docker_machine_spot_price_bid" {
   default     = "0.1"
 }
 
+variable "docker_machine_user" {
```
Comment (ToROxI) [inline variables.tf]: This variable has no any references in code

File: variables.tf [hunk 6]
```diff
@@ -178,6 +161,23 @@ variable "tags" {
   default     = {}
 }
 
+variable "vpc_id" {
+  description = "The VPC that is used for the instances."
+  type        = "string"
+}
+
+variable "availability_zone_proxy" {
```
Comment (ToROxI) [inline variables.tf]: This variable has no any references in code

File: main.tf (hunk 2 above)
Comment (tsimafeikhrol) [reply main.tf]:
```suggestion
    self        = "true"
```

Comment (tsimafeikhrol) [reply main.tf]: Removed var.bastion_ip variable. By default access available only from SG members

File: variables.tf (hunk 4 above)
Comment (tsimafeikhrol) [reply variables.tf]: variable "instance_type" is used in runner-from-prebacked-ami.tf

File: variables.tf (hunk 5 above)
Comment (tsimafeikhrol) [reply variables.tf]: fixed

File: variables.tf (hunk 6 above)
Comment (tsimafeikhrol) [reply variables.tf]: fixed

File: variables.tf (hunk 4 above)
Comment (ToROxI) [reply variables.tf]:
```
tf-module-aws-gitlab-runner(develop) » grep -ri 'cache_user' *                                                                                                               

README.md:| cache_user                    | User name of the user to create to write and read to the s3 cache.                                                  | string |   `cache_user`   |    no    |
variables.tf:variable "cache_user" {
variables.tf:  default     = "cache_user"
```

Comment (tsimafeikhrol) [reply variables.tf]: Fixed

File: variables.tf (hunk 3 above)
Comment (tsimafeikhrol) [reply variables.tf]: Refactored: getting current aws_regoin from data now

File: bucket.tf (hunk 1 above)
Comment (ToROxI) [reply bucket.tf]: any comments?
My propose caused to help AWS Console users to explicitly identify bucket purpose.

Comment (tsimafeikhrol) [reply bucket.tf]: Agree. Done.

File: README.md [hunk 7]
```diff
@@ -35,20 +35,10 @@ resource "aws_iam_service_linked_role" "autoscaling" {
...
-runner_token  = "RUNNER_TOKEN"
-```
-
 ### Conditional creation
 ```hcl
-module "gitlab-runner" {
-  source  = "github.com/lean-delivery/tf-module-aws-gitlab-runner"
-  version = "0.0.0"
+ module "gitlab-runner" {
+  source = "git::https://github.com/lean-delivery/tf-module-aws-gitlab-runner.git?ref=master"
```
[the remark this answers is not in the document]
Comment (tsimafeikhrol) [reply README.md]: Changed

**Pull request merged**